satpy/composites/__init__.py

```diff
@@ -678,7 +688,7 @@ def add_bands(data, bands):
         data['bands'] = new_bands
         data.attrs['mode'] = mode
     # Add alpha band
-    if 'A' not in data['bands'].data and 'A' in bands.data:
+    if ('A' not in data['bands'].data and 'A' in bands.data) or ('A' not in data['bands'].data):
```

Comment:
Unless I'm reading this wrong, is the first expression the same as the last one? If so then logically this just becomes `if 'A' not in data['bands'].data:` which I don't think is what is wanted, right?

Reply:
Oops that's for the previous structure. I'll fix it.